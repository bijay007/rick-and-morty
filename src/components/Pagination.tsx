import React from 'react';
import {
  View, Text, StyleSheet, TouchableOpacity, FlatList,
} from 'react-native';

const Pagination = (props) => {
  const setCurrentPage = (page) => props.onPress(page);
  const createPageIndex = () => {
    const pageIndex = [];
    const { totalPages } = props;
    const { currentPage } = props;
    for (let index = 1; index <= totalPages; index++) {
      pageIndex.push(
        <TouchableOpacity onPress={() => setCurrentPage(index)}>
          <View style={styles.page_number}>
            <Text style={
              index === currentPage
                ? styles.active_page
                : null
            }
            >
              {index}
            </Text>
          </View>
        </TouchableOpacity>,
      );
    }
    return pageIndex;
  };

  return (
    <View style={styles.pagination}>
      <FlatList
        horizontal
        showsHorizontalScrollIndicator={false}
        data={createPageIndex()}
        keyExtractor={(item, index) => index.toString()}
        renderItem={({ item }) => item}
        removeClippedSubviews
      />
    </View>
  );
};

export default Pagination;

const styles = StyleSheet.create({
  pagination: {
    flex: 1,
    padding: 2,
    height: 50,
  },
  page_number: {
    flex: 1,
    width: 50,
    paddingTop: 5,
    alignItems: 'center',
    justifyContent: 'center',
  },
  active_page: {
    color: '#3be1c5',
    textShadowRadius: 2,
    textShadowColor: '#c91f37',
    transform: [{ scale: 1.4 }],
  },
});
