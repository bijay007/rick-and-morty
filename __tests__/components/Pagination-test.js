import React from 'react';
import * as renderer from 'react-test-renderer';
import Pagination from '../../src/components/Pagination';

const fakeProps = {
  totalPages: 0,
  currentPage: 0,
  character: '',
  onPress: () => {},
};

describe('Pagination', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });
  it('renders correctly', async () => {
    const props = fakeProps;
    const tree = await renderer.create(<Pagination props={props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should be passed props object with 4 properties', async () => {
    const tree = await renderer.create(<Pagination />).toTree();
    const { propTypes } = tree.type;
    const fakePropsKeys = Object.keys(fakeProps).sort();
    const propTypesKeys = Object.keys(propTypes).sort();
    expect(propTypesKeys).toEqual(fakePropsKeys);
  });
  it('should return single <ScrollView> component', async () => {
    const tree = await renderer.create(<Pagination />).toJSON();
    expect(tree.children.length).toBe(1);
    expect(tree.type).toBe('RCTScrollView');
  });
});
