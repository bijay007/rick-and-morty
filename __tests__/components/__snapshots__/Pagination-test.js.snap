// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#505050",
      "flex": 1,
      "height": 50,
      "paddingVertical": 5,
    }
  }
>
  <View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
